cpp/log/cert.cc

```diff
@@ -671,6 +696,31 @@ util::Status Cert::PublicKeySha256Digest(string* result) const {
 }
 
 
+StatusOr<string> Cert::SPKI() const {
+  if (!IsLoaded()) {
+    LOG(ERROR) << "Cert not loaded";
+    return util::Status(Code::FAILED_PRECONDITION, "Cert not loaded");
+  }
+
+  unsigned char* der_buf(nullptr);
+  int der_length =
+      i2d_X509_PUBKEY(X509_get_X509_PUBKEY(x509_.get()), &der_buf);
```

Comment:
`X509_get_X509_PUBKEY`, that never returns null? Or, if it does, `i2d_X509_PUBKEY` can handle it?